### ISSUE TITLE
Visual editor numbers chunks incorrectly if chunk nested in callout

### ISSUE THREAD

**Issue by hadley** [rstudio/rstudio#11410]
### System details

    RStudio Edition : 
    RStudio Version : 
    OS Version      : 
    R Version       : 

### Steps to reproduce the problem

Create a quarto doc with the following contents and switch to the visual editor:

````
```{r}
# chunk 1
```

::: callout-note
```{r}
# chunk 2
```
:::

```{r}
# chunk 3
```

```{r}
# chunk 4
```
````

### Describe the problem in detail

Attempt to navigate to chunk 3

<img width="147" alt="Screen Shot 2022-06-10 at 08 49 29" src="https://user-images.githubusercontent.com/4196/173080004-b9ed6e48-275e-4f9e-96b8-0259ac5abc47.png">

### Describe the behavior you expected

Navigator should list 4 chunks, and clicking chunk 3 should take you to chunk 3.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by timtmok** [comment]
I think it's the document outline parsing that doesn't handle the embedded R chunk. It doesn't appear to be an issue in source mode since the callout isn't parsed in the outline. In visual mode, the callout is considered a top level node and the contents aren't parsed to create the outline.

I would even expect the div-callout to show up in the document outline at the very least since the chunk is rendered. Adding the R chunk as a sub-node would also make it easy to navigate to. Although, I think matching what the source mode shows would make the most sense (show all R chunks in the outline).

**Comment by timtmok** [comment]
It appears that this should be handled when gathering the nodes for the outline: https://github.com/rstudio/rstudio/blob/main/src/gwt/panmirror/src/editor/src/api/outline.ts#L137-L147

I don't see anything odd when building the `DocumentOutlineWidget` to exclude the node. Debugging so far has shown that all the chunks are added to the outline. So there's something I'm not seeing where it excludes the nested chunk from visual mode.

@jjallaire Does this seem like the right track?

**Comment by ronblum** [comment]
Confirmed fixed in RStudio Desktop 2022.11.0+91 with R 4.2.0/1 on
- MacOS 12.5 <kbd>Ctrl-Opt-Shift-J</kbd>
- Windows 11 <kbd>Alt-Shift-J</kbd>

Verified that all four chunks are displayed, and that jumping is correct to each of the four.
